src/proto/feature_flag.proto

```diff
@@ -0,0 +1,33 @@
+syntax = "proto2";
+option optimize_for = LITE_RUNTIME;
+
+package otbr;
+
+// Feature flag list message defines the list of feature flags to be used
+// on OpenThread.
+//
+/////////////         Definition Policy Guide            /////////////
+// * Every field is optional with a default value. The default value should
+//   allow the feature to behave safe when the upper layer wants to rollback
+//   or doesn't have the flag value (e.g., due to some network config download
+//   error, API version mismatch, etc.). This makes sure that the system can
+//   always rollback to safe status, even in the edge case.
+// * When a feature finishes the rollup and becomes stable, the clear up process
+//   should update the feature flag default value with the rolluped value.
+
+/////////////  Feature Flag & API Evolving Policy Guide  /////////////
+// * Add a new feature flag: add a new field in the proto.
+// * Remove feature flag: mark the field as deprecated. Do NOT delete the flag.
+// * Rename feature flag: if the feature’s semantics are not changed, it is
+//   allowed to rename the field (the tag should not be changed). If the feature’s
+//   semantics are changed, deprecate the field and add a new one.
+// * Divide the flag into 2 flags: deprecate the old feature field, and add 2
+//   new feature fields.
+// * API Compatibility: If the OTBR API is older than the upper layer
+//   API, there are unrecognized fields in the proto and will be ignored. If the
+//   OTBR API is newer than the upper layer API, the field’s default value will be
+//   used as the feature flag value.
```

Comment:
It may help to define the policy related to deprecated flags.
- If a newer OTBR API deprecates a previously defined flag and upper layer API is older (not aware of the deprecation) and uses the flag, what behavior do we expect from OTBR?

Reply:
If a newer OTBR API deprecates a previously defined flag and upper layer API is older (not aware of the deprecation) and uses the flag, the flag takes no effect on OTBR as OTBR API deprecates it. Added this policy in the proto description.

Reply:
Should we require some sort of backward compatibility (particuallrly when an existing flag is divided into multiple flags).

```
// * Divide the flag into 2 flags: deprecate the old feature field, and add 2
//   new feature fields.
```

For example
- Suppose in future `enable_nat64` is deprecated and replaced with two related flags: `enable_nat64_prefix_manager` and `enable_nat64_translator`. 
- Older user not aware of the change, may still use `enable_nat64` expecting it would control nat64 behavior
- I think the new OTBR implementation should try to remain backward compatible and support the deprecated `enable_nat64` flag. 
- Maybe it can disable/enable both `prefix_manager` and `translator` if `enable_nat64` is used.

Thoughts?

Reply:
1. Thanks for the good use case example on enable_nat64. I added the point 3 to the description in proto file to support the use case.
2. The proto file is the ground truth of the flag status. For people who rely on these feature flags, old and new user are suggested to go to the proto file and be aware of the change when they make a new release (normally upper layer and this OTBR release together on the device, so that their version is kept in sync).
3. If feature owner wants to keep the deprecated flag backward compatible, he need to specify: what is the behavior if no/portion/all deprecated_flag & its_new_flags exist.
4. If feature owner wants to support the backward_compatible enable_nat64, he shall specify for proto's enable_nat64 field: what is the behavior if no/portion/all enable_nat64, translator, and prefix_manager flags exist.